utils/FileHandler.cpp

```diff
@@ -0,0 +1,291 @@
+#include "FileHandler.hpp"
+
+namespace jbrush {
+
+Vector<Shape*> FileHandler::loadShapesFromFile(const QString& directory)
+{
+    // Throws: file didn't open
+    try
+    {
+        // Get the file at the current directory
+        // and make a text stream to operate on it
+        QFile file(directory);
+        QTextStream qin(&file);
+
+        Vector<Shape*> shapesToReturn;  // Vector of shapes to be returned by the function
+        Shape* shapeExtracted;  // Shape extracted from the file
+        QString firstLine;  // Stores the first line of the file
+
+
+        if(!file.open(QIODevice::ReadOnly))
+        {
+           // If the file didn't open, throw an exception
+           throw BadFile(file.fileName());
+        }
+
+        // Expect an empty line at the beginning of the file
+        firstLine = qin.readLine();
+        if(!firstLine.isEmpty())
+        {
+            throw BadFileFormat(30, "loadShapesFromFile", "empty line", firstLine);
+        }
+
+        // Push each shape extracted back into the vector until we get to the end of the file
+        while(!qin.atEnd())
+        {
+            // Extract shape throws if the shape could not be identified
+            shapeExtracted = extractShape(qin);
+            shapesToReturn.push_back(shapeExtracted);
+        }
+
+        // Close the file when loading is finised
+        file.close();
+
+        return shapesToReturn;
+    }// end try
+    catch (BadFile& exception) {
```

Comment:
Umm, try having 1 catch block since they all derive from ``GeneralException`` you might be able to use polymorphism. Oh and make sure it works because I have no idea.
```suggestion
    catch(GeneralException& exception) {
```